src/components/MasterDetails.js

```diff
@@ -137,6 +137,31 @@ export default class MasterDetails extends React.Component<MasterDetailsDefaultP
   }
   state: MasterDetailsState;
 
+  componentWillReceiveProps(newProps: MasterDetailsProps) {
+    const isRowPresent = newProps.rows.find((row) => {
+      return (row.id === this.state.detailsRowId);
+    });
+    if (!isRowPresent) {
+      if (this.props.multiSelect) {
+        const oldPosition = this.state.selectedRows.indexOf(this.state.detailsRowId);
+        if (oldPosition >= 0) {
+          // If it was in the previous array, remove it...
+          this.state.selectedRows.splice(oldPosition, 1);
```

Comment:
Don't forget that we cannot modify the state like this... We need to call this.setState() with the new state... We should make a copy of the old selectedRows and modify it and then include it in the setState call (together with the updated detailsRowId).